plugins/trace_api_plugin/request_handler.cpp

```diff
@@ -26,105 +26,168 @@ namespace {
       return result;
 
    }
-
-   fc::variants process_actions(const std::vector<action_trace_v0>& actions, const data_handler_function& data_handler, const yield_function& yield ) {
+   fc::variants process_actions(trace_api_type_enum  e_action_trace, const vec_action_trace_variant  && actions, const data_handler_function_v0& data_handler_v0, const data_handler_function_v1 & data_handler_v1,  const return_handler_function & return_handler, const yield_function& yield ) {
       fc::variants result;
       result.reserve(actions.size());
 
       // create a vector of indices to sort based on actions to avoid copies
       std::vector<int> indices(actions.size());
       std::iota(indices.begin(), indices.end(), 0);
-      std::sort(indices.begin(), indices.end(), [&actions](const int& lhs, const int& rhs) -> bool {
-         return actions.at(lhs).global_sequence < actions.at(rhs).global_sequence;
+      std::sort(indices.begin(), indices.end(), [e_action_trace, &actions](const int& lhs, const int& rhs) -> bool {
+         return  e_action_trace == e_action_trace_v0 ? std::get<action_trace_v0>(actions.at(lhs)).global_sequence < std::get<action_trace_v0>(actions.at(rhs)).global_sequence :
+		 	std::get<action_trace_v1>(actions.at(lhs)).global_sequence < std::get<action_trace_v1>(actions.at(rhs)).global_sequence;
       });
 
       for ( int index : indices) {
          yield();
 
          const auto& a = actions.at(index);
-         auto action_variant = fc::mutable_variant_object()
-               ("global_sequence", a.global_sequence)
-               ("receiver", a.receiver.to_string())
-               ("account", a.account.to_string())
-               ("action", a.action.to_string())
-               ("authorization", process_authorizations(a.authorization, yield))
-               ("data", fc::to_hex(a.data.data(), a.data.size()));
-
-         auto params = data_handler(a, yield);
+         auto action_variant =  e_action_trace ==  e_action_trace_v0?   fc::mutable_variant_object()
```

Comment:
To remove a lot of the duplicated code and runtime performance hit.  These can become simple `if constexpr (std::is_same_v<ActionTrace, action_trace_v0>)` and return that.

Reply:
if constexpr (std::is_same_v<ActionTrace, action_trace_v0>)  used,  old code remove.